File: frontend/demo/component/side-nav/side-nav-labelled.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import '@vaadin/side-nav';
import '@vaadin/icon';
import '@vaadin/icons';
import '@vaadin/vertical-layout';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('side-nav-labelled')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <div class="side-nav-sample">
        <!-- tag::snippet[] -->
        <vaadin-vertical-layout theme="spacing">
  
          <vaadin-side-nav style="width:100%">
            <span slot="label">Messages</span>
            <vaadin-side-nav-item path="/inbox">
              <vaadin-icon icon="vaadin:inbox" slot="prefix"></vaadin-icon>
              Inbox
            </vaadin-side-nav-item>
            <vaadin-side-nav-item path="/sent">
              <vaadin-icon icon="vaadin:paperplane" slot="prefix"></vaadin-icon>
              Sent
            </vaadin-side-nav-item>
            <vaadin-side-nav-item path="/trash">
              <vaadin-icon icon="vaadin:trash" slot="prefix"></vaadin-icon>
              Trash
            </vaadin-side-nav-item>
          </vaadin-side-nav>

          <vaadin-side-nav collapsible style="width:100%">
            <span slot="label">Admin</span>
            <vaadin-side-nav-item path="/users">
              <vaadin-icon icon="vaadin:group" slot="prefix"></vaadin-icon>
              Users
            </vaadin-side-nav-item>
            <vaadin-side-nav-item path="/permissions">
              <vaadin-icon icon="vaadin:key" slot="prefix"></vaadin-icon>
              Permissions
            </vaadin-side-nav-item>
          </vaadin-side-nav>

        </vaadin-vertical-layout>
        <!-- end::snippet[] -->
      </div>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/sidenav/SideNavLabelled.java
package com.vaadin.demo.component.sidenav;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.component.sidenav.SideNav;
import com.vaadin.flow.component.sidenav.SideNavItem;
import com.vaadin.flow.router.Route;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.component.sidenav.views.InboxView;
import com.vaadin.demo.component.sidenav.views.PermissionsView;
import com.vaadin.demo.component.sidenav.views.SentView;
import com.vaadin.demo.component.sidenav.views.TrashView;
import com.vaadin.demo.component.sidenav.views.UsersView;

@Route("side-nav-labelled")
public class SideNavLabelled extends Div {

    public SideNavLabelled() {

        // tag::snippet[]
        SideNav messagesNav = new SideNav();
        messagesNav.setLabel("Messages");
        messagesNav.addItem(new SideNavItem("Inbox", InboxView.class, VaadinIcon.INBOX.create()));
        messagesNav.addItem(new SideNavItem("Sent", SentView.class, VaadinIcon.PAPERPLANE.create()));
        messagesNav.addItem(new SideNavItem("Trash", TrashView.class, VaadinIcon.TRASH.create()));

        SideNav adminNav = new SideNav();
        adminNav.setLabel("Admin");
        adminNav.setCollapsible(true);
        adminNav.addItem(new SideNavItem("Users", UsersView.class, VaadinIcon.GROUP.create()));
        adminNav.addItem(new SideNavItem("Permissions", PermissionsView.class, VaadinIcon.KEY.create()));

        // end::snippet[]

        VerticalLayout navWrapper = new VerticalLayout(messagesNav, adminNav);
        navWrapper.setSpacing(true);
        navWrapper.setSizeUndefined();
        messagesNav.setWidthFull();
        adminNav.setWidthFull();
        add(navWrapper);

        messagesNav.getElement().executeJs("window.patchSideNavNavigation(this);"); // hidden-source-line
        adminNav.getElement().executeJs("window.patchSideNavNavigation(this);"); // hidden-source-line

        this.addClassName("side-nav-sample");
    }

    public static class Exporter extends DemoExporter<SideNavLabelled> { // hidden-source-line
    } // hidden-source-line
}
